api/v3/Activity.php

```diff
@@ -392,18 +392,12 @@ function civicrm_api3_activity_get($params) {
  * @return array
  *   new activities list
  */
-function _civicrm_api3_activity_get_formatResult($params, $activities) {
+function _civicrm_api3_activity_get_formatResult($params, $activities, $options) {
   if (!$activities) {
     return $activities;
   }
 
-  $returns = CRM_Utils_Array::value('return', $params, array());
-  if (!is_array($returns)) {
-    $returns = str_replace(' ', '', $returns);
-    $returns = explode(',', $returns);
-  }
-  $returns = array_fill_keys($returns, 1);
-
+  $returns = $options['return'];
```

Comment:
Yes, unfortunately there's a weird quirk where the `_civicrm_api3_get_options_from_params` function only supports the legacy `return.foo` syntax under certain circumstances so that's why I could take out the above lines but had to leave in the below.